src/fklearn/causal/cate_learning/meta_learners.py

```diff
@@ -264,3 +260,195 @@ def p(new_df: pd.DataFrame) -> pd.DataFrame:
+
+
+def _get_learners(
+    df: pd.DataFrame,
+    unique_treatments: list,
+    control_learner: Callable,
+    control_name: str,
+    treatment_learner: Callable,
+    treatment_col: str,
+) -> dict:
+    learners: Dict[str, Callable] = {}
+
+    learner_fcn, _, _ = _get_model_fcn(
+        df, treatment_col, control_name, control_learner
+    )
+    learners[control_name] = learner_fcn
+
+    for treatment_name in unique_treatments:
+        learner_fcn, _, _ = _get_model_fcn(
+            df, treatment_col, treatment_name, treatment_learner
+        )
+        learners[treatment_name] = learner_fcn
+
+    return learners
+
+
+@curry
+def causal_t_classification_learner(
+    df: pd.DataFrame,
+    treatment_col: str,
+    control_name: str,
+    prediction_column: str,
+    learner: Callable,
+    treatment_learner: Callable = None,
+    learner_transformers: List[LearnerFnType] = None,
+) -> LearnerReturnType:
+    """
+    Fits a Causal T-Learner classifier. T-Learner is a meta-learner which learns the
+    Conditional Average Treatment Effect (CATE) through the use of multiple models,
+    one for each treatment. Each model is fitted in a subset of the data, according
+    to the treatment. The CATE $\tau$ is defined as
+    $\tau(x_{i}) = M_{1}(X=x_{i}, T=1) - M_{0}(X=x_{i}, T=0)$, being $M_{1}$ a model
+    fitted with treatment data and $M_{0}$ a model fitted with control data, and
+    they can be a Machine Learning Model such as a LightGBM Classifier and $x_{i}$
+    the feature set of sample $i$.
+
+    References:
+    [1] https://matheusfacure.github.io/python-causality-handbook/21-Meta-Learners.html
+    [2] https://causalml.readthedocs.io/en/latest/methodology.html
+
+    Parameters
+    ----------
+
+    df : pd.DataFrame
+        A Pandas' DataFrame with features and target columns.
+        The model will be trained to predict the target column
+        from the features.
+
+    treatment_col: str
+        The name of the column in `df` which contains the names of
+        the treatments or control to which each data sample was subjected.
+
+    control_name: str
+        The name of the control group.
+
+    prediction_column : str
+        The name of the column with the predictions from the provided learner.
+
+    learner: LearnerMutableParametersFnType
+        A fklearn classification learner function.
+
+    treatment_learner: LearnerFnType
+        An optional fklearn classification learner function.
+
+    learner_transformers: list
+        A list of fklearn transformer functions to be applied after the learner and before estimating the CATE.
+        This parameter may be useful, for example, to estimate the CATE with calibrated classifiers.
+    """
+
+    control_learner = copy.deepcopy(learner)
+
+    if treatment_learner is None:
+        treatment_learner = copy.deepcopy(learner)
+
+    control_features = _get_learner_features(control_learner)
+    treatment_features = _get_learner_features(treatment_learner)
```

Comment:
Since we are using fklearn LearnerFnType learners, which are already curried with the features, why do we need to re-input them? In the S-learner we did that because we were creating a new feature (T). I am not seeing the reason here. Is there a particular reason?